src/main/scala/com.snowplowanalytics.snowplow.micro/InMemoryStorage.scala

```diff
@@ -10,50 +10,37 @@
 
 package com.snowplowanalytics.snowplow.micro
 
+import cats.effect.IO
+
 /** In-memory cache containing the results of the validation (or not) of the tracking events.
   * Good events are stored with their type, their schema and their contexts, if any,
   * so that they can be quickly filtered.
   * Bad events are stored with the error message(s) describing what when wrong.
   */
-private[micro] class ValidationCache(maxEvents: Option[Int]) {
-  import ValidationCache._
+private[micro] class InMemoryStorage extends EventStorage {
+  import InMemoryStorage._
 
   protected var good = List.empty[GoodEvent]
   private object LockGood
   protected var bad = List.empty[BadEvent]
   private object LockBad
 
-  /** Compute a summary with the number of good and bad events currently in cache. */
-  private[micro] def getSummary(): ValidationSummary = {
-    val nbGood = LockGood.synchronized {
-      good.filterNot(_.incomplete).size
-    }
-    val nbBad = LockBad.synchronized {
-      bad.size
-    }
-    ValidationSummary(nbGood + nbBad, nbGood, nbBad)
-  }
-
   /** Add a good event to the cache. */
-  private[micro] def addToGood(events: List[GoodEvent]): Unit =
+  def addToGood(events: List[GoodEvent]): IO[Unit] = IO.delay {
```

Comment:
```suggestion
  override def addToGood(events: List[GoodEvent]): IO[Unit] = IO.delay {
```
It's kind of documentation for other developers  to add `override` when implementing functions from the `trait` so that we can easily differentiate with functions that are not from the `trait` (same for `addToBad` and `reset`)